### DISCUSSION TITLE
have the environment variable as optional

### DISCUSSION THREAD

**Discussion by dagap** [facebookresearch/hydra#2104] [Q&A] [answered]
I am using a third-party SW which has a trial version when the license key is not specified. In my config file, I have the following:

```
licence: ${env:SERVER_LICENSE}
```
Now if this license env variable does not exist, my code fails with:

```
Environment variable 'SERVER_LICENSE' not found
```
Is there a way to allow this to be None or not existing from the config file? I know I can probably set it in the config in case it is missing manually but that is very hacky and I have various alternate tools integrated which does not have this license files requirement and I will have top make this fix for this one tool.

**Comment by Jasha10** [comment, marked as answer]
Please give the following a try:
```
licence: ${oc.env:SERVER_LICENSE,null}
```

Note that `env` is deprecated in favor of `oc.env`. See the docs on the [`oc.env`](https://omegaconf.readthedocs.io/en/latest/custom_resolvers.html#oc-env) resolver.
